### PR TITLE
ref(ownership): refactor post_process telemetry

### PR DESCRIPTION
**Pull request by JoshFerge**
I was finding this code very hard to read with all of the telemetry context creations and indentation. No behavior changes intended.

- [x] Taking advantage of `metrics.wraps` and `sentry_sdk.trace` we can get rid of most of these.
- [x] Additionally, i've removed the spans around the caching operations as all of our cache get / sets get instrumented with spans anways.
- [x] Utilize return early pattern to prevent more nesting
- [x] we don't need a wrapper to try / except the whole function as post process already does this https://github.com/getsentry/sentry/blob/020c9f4a668fdf5adf64791efc856384db4cde87/src/sentry/tasks/post_process.py#L692

### CODE REVIEW

File: src/sentry/tasks/post_process.py [hunk 1]
```diff
@@ -204,109 +205,87 @@ def should_issue_owners_ratelimit(project_id: int, group_id: int, organization_i
...
-                        else:
-                            handle_invalid_group_owners(group)
+    if killswitch_matches_context(
+        "post_process.get-autoassign-owners",
+        {
+            "project_id": project.id,
+        },
+    ):
+        # see ProjectOwnership.get_issue_owners
+        issue_owners: Sequence[tuple[Rule, Sequence[Team | RpcUser], str]] = []
```
Comment (snigdhas) [inline src/sentry/tasks/post_process.py:268]: do we need to go through `handle_invalid_group_owners` in this branch? the old code without the early-return would eventually call `handle_invalid_group_owners(group)`.

Comment (snigdhas) [reply src/sentry/tasks/post_process.py:268]: although imo if the killswitch is set, returning without doing anything seems like the right call

Comment (JoshFerge) [reply src/sentry/tasks/post_process.py:268]: yeah, this is a fair point. it is a behavior change. i'll modify the function to call `handle_invalid_group_owners`. i feel suspicious about even hitting that if the killswitch is activated, but will keep the same.

Comment (JoshFerge) [reply src/sentry/tasks/post_process.py:268]: went ahead and changed that else back

**Review by snigdhas — APPROVED** [review]

**Pull request merged**